rholang/src/main/scala/coop/rchain/rholang/ResourceTools.scala

```diff
@@ -1,45 +1,28 @@
 package coop.rchain.rholang
 import java.io.File
 import java.nio.file.{Files, Path}
-import cats.implicits._
-import cats.{FlatMap, MonadError}
+
+import cats.effect.Resource
+import cats.Applicative
+import cats.effect.ExitCase.{Canceled, Completed, Error}
 import com.typesafe.scalalogging.Logger
 
 import scala.reflect.io.Directory
 
 object ResourceTools {
-  def withResource[F[_]: FlatMap, T, R](
-      acquire: => F[R],
-      release: (R, Option[Throwable]) => F[Unit]
-  )(job: R => F[T])(implicit F: MonadError[F, Throwable]): F[T] =
-    for {
-      resource <- acquire
-      attempt  <- job(resource).attempt
-      result <- attempt.fold(
-                 ex => release(resource, Some(ex)) *> F.raiseError(ex),
-                 r => release(resource, None) *> F.pure(r)
-               )
-    } yield result
-
   val logger: Logger = Logger(this.getClass.getName.stripSuffix("$"))
 
-  def withTempDir[F[_], A](
-      prefix: String
-  )(job: Path => F[A])(implicit F: MonadError[F, Throwable]): F[A] = {
-    def deleteTempDir(path: Path, maybeError: Option[Throwable]): F[Unit] =
-      F.pure(maybeError match {
-        case None => new Directory(new File(path.toString)).deleteRecursively()
-        case Some(ex) =>
-          logger
-            .error(
-              s"Exception thrown while using the tempDir '$path'. Temporary dir NOT deleted.",
-              ex
-            )
-      })
-
-    withResource[F, A, Path](
-      F.pure(Files.createTempDirectory(prefix)),
-      deleteTempDir
-    )(job)
-  }
+  def mkTempDir[F[_]: Applicative](prefix: String): Resource[F, Path] =
+    Resource.makeCase(Applicative[F].pure(Files.createTempDirectory(prefix)))(
+      (path, exitCase) =>
+        Applicative[F].pure(exitCase match {
+          case Error(ex) =>
+            logger
+              .error(
+                s"Exception thrown while using the tempDir '$path'. Temporary dir NOT deleted.",
```

Comment:
Are you sure? Even if exception is thrown during the usage we should delete the directory.

Reply:
This is to be discussed. Let's get @ArturGajowy in this discussion as well. He suggested this approach and I think it's a good idea not to delete temp files when the test fails because there may be debugging information in those temp files which can be helpful.

Reply:
How would you use the residual directory for debugging? I am just asking b/c I have never found a need for that and we don't have a tooling.

Reply:
That's something we should figure out how to do.

Reply:
I'd rather have people do rm /tmp/* from time to time than say 'oh, this blew up on me. Do I debug this? No, there's not trace after that... Rerun, tests green, my job is done here'.